client/src/components/overview/ProductOverview.jsx

```diff
@@ -118,16 +161,17 @@ class ProductOverview extends React.Component {
   setStyleById(id) {
     id = parseInt(id);
     if (this.state.selectedStyle.style_id !== id) {
-      const style = this.getStyleById(id);
+      const style = this.getStyleById(id, this.state.styles);
```

Comment:
I know you're currently refactoring, but unless you have some surprises coming, `styles` does not seem like something that React state is set up to deal with. Keep in mind, setState is a rather unpredictable async operation. Unless you need to trigger renders, using the cache or regular class variables (`this.styles`) and regular assignment might be a better/less error prone way to keep track of things.

Reply:
I need to look more broadly later to be certain, but this was necessary once using the cache, as I need to trigger a render at some unknown later time. Before I made this change, the component didn't update later in the app render cycle once data was available.